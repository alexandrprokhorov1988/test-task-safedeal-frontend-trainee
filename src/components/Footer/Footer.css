.footer {
  margin: 0 auto;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 95%;
  max-width: 728px;
  border-top: 1px solid #CCCCCC;
}

.footer__text {
  margin: 0;
  color: #CCCCCC;
  font-size: 14px;
  line-height: 16px;
  padding: 20px;
}
