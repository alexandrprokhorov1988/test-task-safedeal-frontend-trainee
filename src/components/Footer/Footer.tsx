import React from 'react';
import style from './Footer.module.css';

/**
 * Footer component.
 */
const Footer: React.FC = () => {

  return (
    <footer className={style.footer}>
      <p className={style.text}>Â© 2018-2019-2021</p>
    </footer>
  );
};

export default Footer;
