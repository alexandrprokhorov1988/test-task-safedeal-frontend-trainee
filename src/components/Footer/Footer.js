import React from 'react';
import './Footer.css';

function Footer() {

  return (
    <footer className="footer">
      <p className="footer__text">Â© 2018-2019-2021</p>
    </footer>
  );
}

export default Footer;
