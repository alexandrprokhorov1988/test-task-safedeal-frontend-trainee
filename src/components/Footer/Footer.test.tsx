import React from 'react';
import {render, unmountComponentAtNode} from "react-dom";
import renderer from 'react-test-renderer';
import {act} from "react-dom/test-utils";
import Footer from './Footer';

let container:any = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('Footer component', () => {
  it("Рендер footer с текстом", () => {
    act(() => {
      render(<Footer/>, container);
    });
    expect(container.textContent).toBe("© 2018-2019-2021");
  });

  it('Рендер footer snapshot', () => {
    const tree = renderer
      .create(<Footer/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
