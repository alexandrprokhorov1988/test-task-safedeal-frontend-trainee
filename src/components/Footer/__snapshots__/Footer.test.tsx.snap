// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Рендер footer snapshot 1`] = `
<footer
  className="footer"
>
  <p
    className="footer__text"
  >
    © 2018-2019-2021
  </p>
</footer>
`;
